cmd/snap/cmd_disconnect.go

```diff
@@ -30,13 +30,17 @@ import (
 
 type cmdDisconnect struct {
 	waitMixin
+	Forget      bool `long:"forget"`
 	Positionals struct {
 		Offer disconnectSlotOrPlugSpec `required:"true"`
 		Use   disconnectSlotSpec
 	} `positional-args:"true"`
 }
 
 var shortDisconnectHelp = i18n.G("Disconnect a plug from a slot")
+
+// XXX we might give hint about --forget here once old connections
+// are made discoverable for the user.
```

Comment:
it seems the help of disconnect doesn't mention that disconnection are remembered (vs auto connection),  we should probably expand the help to mention that and clarify what is different wrt that with --forget

Reply:
How about adding something like this below the existing description (new text in italic):

```
...
Disconnects everything from the provided plug or slot.
The snap name may be omitted for the core snap.

_Connections that were created automatically upon snap
installation and then manually disconnected are remembered
internally and will not get re-connected on
snap refresh. The --forget flag may be passed to
disconnect command in order to remove remembered state about
specific connections._
```

CC @degville

Reply:
how about:

When an automatic connection is manually disconnected, its disconnected state is retained after a snap refresh. The --forget flag can be added to the disconnect command to reset this behaviour, and consequently re-enable an automatic reconnection after a snap refresh.

Reply:
Thanks Graham, this sounds good to me!